checker.js

```diff
@@ -15,341 +15,525 @@
-									function (err, stdout, stderr) {
-										debugComp("Visually compared Image #%s: \nstdout: %s \nstderr: %s \nerr: %s", k, stdout, stderr, err);
-
-										if(err) {
-											debugERROR("%s", err.message);
-											return;
-										}
-
-										exec("base64 tmp_comp_imgOrigRep_" + k +".png > tmp_comp_base64_" + k + ".txt", function (err, stdout, stderr) {
-											debugComp("Writing base64 file for Image #%s: \nstdout: %s \nstderr: %s \nerr: %s", k, stdout, stderr, err);
-											if (k == base64ImagesOriginal.length-1) {
-												checkIfAllImagesAreDoneComparing();
-											}
-										});
-									})
-							}, 5);
-						}
-					})
-				}
+							metadata.images[index].compareResults = {
+								differences : result.differences,
+								dimension : result.dimension
+							};
 
-				function checkIfAllImagesAreDoneComparing() {
-					var countFound = 0;
-					for ( var numImage = 0; numImage < differingImagesCount; numImage++ ) {
-						try {
-							var checkIfFileExists = fs.statSync(path.join(process.cwd(), "tmp_comp_base64_" + differingImagePositions[numImage] + ".txt"));
-						}
-						catch (e) {
-							countFound -= 1;
-						}
-						finally {
-							countFound += 1;
+							if (countComparedImages === images.length) {
+								resolve(resultImages);
+							}
 						}
-					}
-					if (countFound == differingImagesCount) {
-
-						debugComp("\tFinished comparison.".magenta)
-						reassembleHTMLfromComparedImagesAndText();
-					}
-					else {
-						setTimeout(function () {
-							checkIfAllImagesAreDoneComparing();
-						}, 20);
-					}
-			}
-				
-				checkIfEncodedImagesAreExtracted();
-			});
+					)
+				}
+			)
 		}
-
-	});
+	)
 }
 
+function reassembleDiffHTML (diffImageBufferArray, textChunkArray, outputName) {
 
-function getContentsOfImageTags(stringifiedHTML) {
+	let reassembledDiffHTMLString = "";
 
-	return stringifiedHTML.match(allImgTagsAsStrings).map(function (finding) {
-		return finding.substr(8, finding.length);
-	});
+	debugReassemble("Piecing together text chunks and images.");
+	textChunkArray.map(
+		function (currentTextChunk, index) {
+			reassembledDiffHTMLString += currentTextChunk
+				+  "<img src=\"data:image/png;base64,"
+				+ toBase64.encode(diffImageBufferArray[index].buffer)
+				+ "\" width=\"672\" />";
+		}
+	);
+	reassembledDiffHTMLString += textChunkArray.pop();
+	debugReassemble("Reassembly done.".green);
+	debugGeneral("Writing result HTML".cyan);
+
+	metadata.resultHTML = reassembledDiffHTMLString;
+
+	try {
+		if(outputName) {
+			fs.writeFileSync("./" + outputName + ".html", reassembledDiffHTMLString);
+		}
+		else {
+			fs.writeFileSync("./result.html", reassembledDiffHTMLString);
+		}
+	}
+	catch(e) {
+		debugERROR("Failed to write result HTML file.".red);
+		metadata.errorsEncountered.push(e);
+		debugERROR(e);
+		return;
+	}
+	debugGeneral("Output files written successfully".green);
+	//debugGeneral(metadata);
+	return metadata;
 }
 
+/*
```

Comment:
Remove such commented out lines in favour of proper tests, or even put them in drafts/concepts for tests within `/test`